src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1688,7 +1688,17 @@ void TorrentHandle::handleTrackerErrorAlert(const lt::tracker_error_alert *p)
 
     m_trackerInfos[trackerUrl].lastMessage = message;
 
-    m_session->handleTorrentTrackerError(this, trackerUrl);
+    // In libtorrent 1.2.x each tracker has multiple local endpoints from which an announce is
+    // attempted. Some endpoints might succeed while others might fail. Emit the signal only
+    // if all endpoints have failed. TrackerEntry::isWorking() returns true if at least
+    // one endpoint works.
+    for (const TrackerEntry &entry : asConst(trackers())) {
```

Comment:
I would use `std::any_of()` here but I don't insist.

Reply:
I changed the if tests a bit to be more efficient. I think I can't use any_of() now.

Reply:
But if you think it can still be used please provide (pseudo) code.